pkg/webhook/policy_benchmark_test.go

```diff
@@ -0,0 +1,301 @@
+func readTemplates(pathList []string) (templates.ConstraintTemplateList, error) {
+	var result templates.ConstraintTemplateList
+	result.Items = make([]templates.ConstraintTemplate, len(pathList))
+	for i, path := range pathList {
+		t := buildTemplatePath(path)
+		yamlString, err := ioutil.ReadFile(t)
+		if err != nil {
+			return templates.ConstraintTemplateList{}, err
+		}
+		cstr := &templv1beta1.ConstraintTemplate{}
+		if err := yaml.Unmarshal(yamlString, cstr); err != nil {
+			return templates.ConstraintTemplateList{}, err
+		}
+		unversioned := templates.ConstraintTemplate{}
+		if err := runtimeScheme.Convert(cstr, &unversioned, nil); err != nil {
+			return templates.ConstraintTemplateList{}, err
+		}
+		result.Items[i] = unversioned
+	}
+	return result, nil
+}
+func buildTemplatePath(path string) string {
+	var b strings.Builder
+	b.WriteString(path)
+	b.WriteString("template.yaml")
+	return b.String()
+}
+func buildConstraintPath(path string) string {
+	var b strings.Builder
+	b.WriteString(path)
+	b.WriteString("constraint.yaml")
+	return b.String()
+}
+func buildExamplePath(path string) string {
+	var b strings.Builder
+	b.WriteString(path)
+	b.WriteString("example.yaml")
+	return b.String()
+}
+
+// readConstraints reads constraints from the pathList
+func readConstraints(pathList []string) ([]unstructured.Unstructured, error) {
+	result := make([]unstructured.Unstructured, len(pathList))
+	for i, path := range pathList {
+		c := buildConstraintPath(path)
+		yamlString, err := ioutil.ReadFile(c)
+		if err != nil {
+			return nil, err
+		}
+		cr := unstructured.Unstructured{}
+		if err := yaml.Unmarshal(yamlString, &cr); err != nil {
+			return nil, err
+		}
+		result[i] = cr
+	}
+	return result, nil
+}
+
+func addTemplates(opa *opa.Client, list templates.ConstraintTemplateList) error {
+	for _, ct := range list.Items {
+		_, err := opa.AddTemplate(context.Background(), &ct)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func addConstraints(opa *opa.Client, list unstructured.UnstructuredList) error {
+	for _, cr := range list.Items {
+		_, err := opa.AddConstraint(context.TODO(), &cr)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// generateConstraints generates M constraints based on representative constraint in crList
+func generateConstraints(M int, crList []unstructured.Unstructured) unstructured.UnstructuredList {
+	result := unstructured.UnstructuredList{}
+	result.Items = make([]unstructured.Unstructured, M)
+	// seed random generator
+	rand.Seed(time.Now().UnixNano())
+	for i := 0; i < M; i++ {
+		r := crList[i%len(crList)]
+		result.Items[i] = *(r.DeepCopy())
+		r.SetName(genRandString(10))
+	}
+	return result
+}
+
+func genRandString(n int) string {
+	out := make([]byte, n)
+	for i := 0; i < n; i++ {
+		c := 'a' + rand.Intn(26)
+		out[i] = byte(c)
+	}
+	return string(out)
+}
+
+// libraryPathList contains a random set of library paths with resources
```

Comment:
This is not currently random, should the docstring be updated? Would we want it random, or would we want consistent test inputs to promote comparability?

Reply:
Docstring should be updated. I was going with consistent test inputs to promote comparability. It would be hard to characterize the results with random inputs